test/generate_dot_test.cc

```diff
@@ -0,0 +1,142 @@
+#include <maliput/routing/routing_constraints.h>
+#include <maliput/utility/generate_dot.h>
+#include <maliput_malidrive/builder/params.h>
+#include <maliput_malidrive/builder/road_network_builder.h>
+#include <maliput_malidrive/loader/loader.h>
+
+namespace maliput {
+namespace test {
+namespace {
+
+static constexpr char kMalidriveResourcesPath[] = DEF_MALIDRIVE_RESOURCES;
+
+class TShapeRoadGenerateDotTest : public ::testing::Test {
+ public:
+  //@{  Tolerances set to match the involved geometries and the parser resolution.
+  static constexpr double kLinearTolerance{1e-6};
+  static constexpr double kAngularTolerance{1e-6};
+  static constexpr double kScaleLength{1.0};
+  //@}
+  static constexpr routing::RoutingConstraints kDefaultRoutingConstraints{};
+  const std::string kTShapeRoadFilePath{std::string(kMalidriveResourcesPath) +
+                                        std::string("/resources/odr/TShapeRoad.xodr")};
+  const api::LaneId kStartLaneId{"0_0_1"};
+  const api::LaneId kEndLaneId{"0_0_1"};
+
+  std::unique_ptr<api::RoadNetwork> road_network_{};
+  std::unique_ptr<routing::Router> router_{};
+  routing::graph::Graph graph_{};
+  std::vector<routing::Route> routes_{};
+
+  void SetUp() override {
+    std::map<std::string, std::string> road_network_configuration;
+    road_network_configuration.emplace(malidrive::builder::params::kRoadGeometryId, "malidrive_rg");
+    road_network_configuration.emplace(malidrive::builder::params::kOpendriveFile, kTShapeRoadFilePath);
+    road_network_configuration.emplace(malidrive::builder::params::kLinearTolerance, std::to_string(kLinearTolerance));
+    road_network_configuration.emplace(malidrive::builder::params::kAngularTolerance,
+                                       std::to_string(kAngularTolerance));
+    road_network_configuration.emplace(malidrive::builder::params::kScaleLength, std::to_string(kScaleLength));
+    road_network_configuration.emplace(malidrive::builder::params::kInertialToBackendFrameTranslation, "{0., 0., 0.}");
+    road_network_ = malidrive::loader::Load<malidrive::builder::RoadNetworkBuilder>(road_network_configuration);
+
+    router_ = std::make_unique<maliput::DistanceRouter>(*road_network_, kLinearTolerance);
+
+    graph_ = routing::graph::BuildGraph(road_network_->road_geometry());
+
+    const api::RoadPosition start(road_network_->road_geometry()->ById().GetLane(kStartLaneId),
+                                  api::LanePosition(1., 0., 0.));
+    const api::RoadPosition end(road_network_->road_geometry()->ById().GetLane(kEndLaneId),
+                                api::LanePosition(10., 0., 0.));
+    routes_ = router_->ComputeRoutes(start, end, kDefaultRoutingConstraints);
+    ASSERT_EQ(1u, routes_.size());
+  }
+};
+
+// Evaluates the graph is correct and the api::Segment containing the routing::Route is marked with the red color.
+TEST_F(TShapeRoadGenerateDotTest, DotGraphWithARouteMarksInRedWhereTheRouteGoesThrough) {
```

Comment:
> DotGraphWithARouteMarksInRedWhereTheRouteGoesThrough

Consider simplifying to be "DotGraphWithRouteHighlightedInRed."

Reply:
[db8afbd](https://github.com/maliput/maliput_integration_tests/pull/85/commits/db8afbd2ce9695f2fe50d579c4132901bcfaf4ea)